File: cli/sp-cli/src/services/purchaseorderservice.ts
import { HttpService, PurchaseOrderService as BasePurchaseOrderService, Party } from 'wt-common-cli';

export class PurchaseOrderService extends BasePurchaseOrderService {
    constructor(httpService: HttpService) {
        super(httpService);
    }

    async verifyPurchaseOrder(purchaseOrderId: string, contentOwnerGln: number, buyer: Party, seller: Party, units: number, price: number, productGtin): Promise<boolean> {
        const resp = await this.httpService.post(`purchaseorders/${purchaseOrderId}/verify`, {
            contentOwnerGln,
            buyer,
            seller,
            units,
            price,
            productGtin,
        });

        return resp.toString() === 'true';
    }
}

File: sp-rest/src/main/java/org/tradefinance/sprest/services/PurchaseOrderService.java
package org.tradefinance.sprest.services;

import org.tradefinance.assets.defs.Party;

public interface PurchaseOrderService extends org.tradefinance.common.services.PurchaseOrderService {
    public boolean verifyPurchaseOrder(String purchaseOrderId, long contentOwnerGln, Party buyer, Party seller, double price, int units, long productGtin) throws Exception;
}
